drivers/storage/partmgr/partition.c

```diff
@@ -173,6 +172,103 @@ PartitionHandleStartDevice(
     return STATUS_SUCCESS;
 }
 
+/**
+ * @brief
+ * Notifies MountMgr to delete all mount points
+ * associated with the given volume.
+ *
+ * @note    This should belong to volmgr.sys and act on a PVOLUME_EXTENSION.
+ **/
+static
+CODE_SEG("PAGE")
+NTSTATUS
+VolumeDeleteMountPoints(
+    _In_ PPARTITION_EXTENSION PartExt)
+{
+    NTSTATUS Status;
+    UNICODE_STRING MountMgr;
+    ULONG InputSize, OutputSize;
+    LOGICAL Retry;
```

Comment:
Interesting, I would never have known about `LOGICAL`. How is `LOGICAL` implemented? In my opinion it would have made more sense to use `ULONG`.

Reply:
`LOGICAL` is a typedef ULONG, in the main DDK (wdm.h) headers. Iirc. it has been introduced as a type during WinXP or 2003. My personal understanding is that it's meant to be used for e.g. booleans TRUE/FALSE when you need to store this in a 4-byte-large value, rather than in the usual 1-byte value (i.e. the `BOOLEAN` type), or for multi-state "booleans", like trooleans etc.
This is why I used it there because for me the Retry values are: 0 (FALSE, default), then 1 (TRUE), then 2+ (abort).